testsuite/elytron/src/test/java/org/wildfly/test/security/common/elytron/CliFragment.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.test.security.common.elytron;
+
+/**
+ * Represents common piece in CLI commands, which can be shared across types.
+ *
+ * @author Josef Cacek
+ */
+public interface CliFragment {
+
+    /**
+     * Generates part of CLI string which uses configuration for this fragment.
+     *
+     * @return part of CLI command
+     */
+    String asString();
```

Comment:
The `asString` makes it IMO more obvious that it's not used just for printing into log. It's not just a textual representation of the data, it also has to have a correct form to be used in CLI command.